Security/Guidelines/OpenID_connect.mediawiki

```diff
@@ -0,0 +1,177 @@
+
+Other tokens can be used, though these do not pertain directly to authentication. These are also often called OAuth2 tokens. This is because OIDC is based on OAuth2 and thus also provides full OAuth2 support. The OAuth2 tokens (Access Token and Refresh Tokens) enable their bearer to access information from other websites and resources (including additional user attributes that may not be passed by the ID token) - but are not required to perform user authentication.
+
+==== OIDC tokens reference table ====
+{| class="wikitable"
+! Token
+! Format
+! Description
+|-
+! '''Access token'''
+| A string containing a unique secret token (like an API key).
+| The Access token has specific permissions and is used to get data from an API. It expire quickly, typically within 24 hours.
+|-
+! '''Refresh token'''
+| A string containing a unique secret token (like an API key).
+| The Refresh token enables it's bearer to request and obtain new Access tokens. These newly obtained access tokens have a subset of the permissions that the Refresh token has. The Refresh token never expires.
+|-
+! '''ID token'''
+| Base64-encoded JSON document ([https://jwt.io/ JWT])
+| The ID token contains information about how and when the user authenticated along with various attributes. ID tokens are created and signed by OpenID Connect Providers (OP) and consumed, verified by web applications authenticating users (RPs).
+|-
+|}
+
+==== Detailed OIDC authentication flow ====
+
+This sequence diagram is useful if you want to understand how OIDC works, or need to modify or implement your own OIDC library.
+
+[[File:Authentication_Sequence_with_OpenID_Connect(1).png|1100px]]
+
+== Implement authentication with OpenID Connect (OIDC) securely in my web applications (RP) ==
+
+=== Session handling ===
+
+The OpenID Connect Provider (OP) typically creates a user session cookie so that it does not need to re-ask the user for their
+credentials too often across different web applications (RP).
+The expiration of the session depends on the OP setup the session and may be forced to expire by the OpenID Connect Provider (OP) sooner than the cookie indicates on the user's browser.
+This allows the OP to forcibly log the user out from the OP point of view. This premature termination of the user’s session with the OP will not, however, end the user’s session on the web application's (RP's) which they’ve logged into.
+
+For that reason, it is important that the web application (RP) respect the following set of rules in regards to session handling:
+
+
+* '''Must''' expire the user session when the ID token expires or sooner (the expiration time is generally a UNIX timestamp attribute named <code>exp</code>).
+* If the user's session is longer than '''15 minutes''', '''must''' re-check/introspect the ID token every ''15 minutes'' or next user request (whichever comes first), to ensure that the user is still valid and has correct permissions.
```

Comment:
Also, in talking to @gdestuynder this is actually 2 different rules depending on if the OP supports `/delegate` or not and should be broken out into two rules.